hlib/python/hlib/frontend/relay_parser.py

```diff
@@ -0,0 +1,696 @@
+                if isinstance(item, str):
+                    if type_dict[item] == Call:
+                        inner_func = bind_env[item][1]
+                        inner_args = bind_env[item][2]
+                        inner_kwargs = bind_env[item][3]
+                        _args[i] = inner_func(*inner_args, **inner_kwargs)
+            _args = tuple(_args)
+            arg_list = []
+            for _var in _args:
+                if _var in params:
+                    arg_list.append(_var)
+                else:
+                    arg_list.append(_var)
+            return _func(*arg_list, **_kwargs)
+    if ntype == Tuple:
+        var = arg[0][0]
+        env = arg[2]
+        tup = env[var][1]
+        dict_type = env[var][2]
+        tup_env = env[var][3]
+        return tuple_extract(tup, dict_type, tup_env)
+    if ntype == Var:
+        name = arg[0][0]
+        return (arg[2])[name]
+    else:
+        print("Type not implemented yet")
+
+
+def isInParams(var, params):
+    if(not isinstance(var, hcl.tensor.Tensor)):
+        return False
+
+    for par in params:
+        isShape = (var.shape == par.shape)
+        isName = (var.name == par.name)
+        isType = (var.dtype == par.dtype)
+        if(isShape and isName and isType):
+            return True
+    return False
+
+
+def get_type(ty, name):
+    if isinstance(ty, TensorType):
+        dtype = ty.dtype
+        size = []
+        for i in ty.shape:
+            size.append(i.value)
+        return hcl.placeholder(tuple(size), name, dtype)
+    elif isinstance(ty, TupleType):
+        t_vars = []
+        for i in range(len(ty.fields)):
+            var_name = name + "_" + str(i)
+            t_vars.append(get_type(ty.fields[i], var_name))
+        return tuple(t_vars)
+    else:
+        pass
+
+# get_Item
+
+
+def get_item(env):
+    tup_type = env[1]
+    if tup_type == Var:
+        tup = list(env[2])
+        index = env[3]
+        item = tup[index]
+        if(isinstance(item, tuple)):
+            name = env[0]
+        else:
+            name = item.name
+        inst_type = {name: Var}
+        inst_env = {name: item}
+    if tup_type == Call:
+        name = env[0]
+        tup = env[2]
+        index = env[3]
+    return item, name, inst_type, inst_env, inst_var
+
+
+def gen_code(item, params, var, type_dict, env):
+    if(debug_mode):
+        print("Item:", item)
+    if(type_dict[item] == Function):
+        if(debug_mode):
+            print("In Func")
+        _var = env[0]
+        _type_dict = env[1]
+        _env = env[2]
+        _params = gen_params(type_dict, env)
+        _size = env[3]
+        f = gen_func(_params, _var, _type_dict, _env)
+        env[item] = f
+        type_dict[item] = Call
+    elif(type_dict[item] == Let):
+        if(debug_mode):
+            print("In Let")
+        _ntype = env[item][0]
+        _bind_var = env[item][1]
+        _var = env[item][2]
+        _dict = env[item][3]
+        _env = env[item][4]
```

Comment:
Some variables seem to be not useful?

Reply:
Let binding is not used for Keras,but it might be useful for other frontends. I'll comment